.github/workflows/lint.yml

```diff
@@ -0,0 +1,42 @@
+name: lint
+
+on:
+    pull_request:
+        branches: [main]
+    workflow_dispatch:
+
+env:
+    CARGO_TERM_COLOR: always
+    ENVIRONMENT: development
+    GH_TOKEN: ${{ github.token }}
+
+jobs:
+    check:
+        runs-on: ubuntu-latest
+        steps:
+            # Checkout the repo
+            - uses: actions/checkout@v3
+
+            - run: mkdir -p protocol/cargo-cache
+            - run: mkdir -p protocol/target
+            - run: mkdir -p node_modules
+            - run: mkdir -p .cache/Cypress
+
+            - name: Restore cache
+              uses: actions/cache/restore@v3
+              with:
+                  path: |
+                      protocol/cargo-cache
+                      protocol/target
+                      node_modules
+                      .cache/Cypress
+                  # note that restoring a cache in github is a pain. The trailing '-' matches any string after the '-', therefore 'abc-' would match a cache named 'abc-1234' or 'abc-5678', etc.
+                  # the problem is 'abc-' will not match a cache named 'abc'! So if you're using wildcard cache name selectors like this, you need a field that changes as the suffix to become the wildcard
+                  # here we're setting the key to an unused cache key so it falls back to the wildcard selector in `restore-keys`
+                  key: some-unused-cache-key
+                  restore-keys: |
+                      project-cache-${{ runner.os }}-${{ runner.arch }}-
+
+            - run: npm install -g npm@8.9
+            - run: npm install
+            - run: npm run lint
```

Comment:
Where is the check to see if the linting is successful?

Reply:
`npm run lint` fails if unsuccessful, exit with code 1